src/legacy/core_plugins/metrics/server/lib/vis_data/helpers/get_es_shard_timeout.js

```diff
@@ -0,0 +1,22 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+export default function getEsShardTimeout(req) {
+  return `${req.server.config().get('elasticsearch.shardTimeout')}ms`;
```

Comment:
if `elasticsearch.shardTimeout` is 0 it means that it's disabled and it's not included in the query request. Take a look at this issue: https://github.com/elastic/kibana/issues/27209 and how it's was fixed here: https://github.com/elastic/kibana/pull/27217

Reply:
Done